read-fonts/src/tables/variations.rs

```diff
@@ -485,16 +498,11 @@ impl Iterator for DeltaRunIter<'_> {
             }
             self.limit = Some(limit - 1);
         }
-        // if no items remain in this run, start the next one.
-        // NOTE: we use `while` so we can sanely handle the case where some
-        // run in the middle of the data has an explicit zero length
-        //TODO: create a font with data of this shape and go crash some font parsers
-        while self.remaining_in_run == 0 {
+        if self.remaining_in_run == 0 {
             let control: u8 = self.cursor.read().ok()?;
             self.value_type = DeltaRunType::new(control);
             self.remaining_in_run = (control & DELTA_RUN_COUNT_MASK) + 1;
         }
```

Comment:
Given the `+ 1`, it looks like encoding an explicit zero length run is impossible, so I removed this loop and comment.